tests/test_json.py

```diff
@@ -188,6 +191,44 @@ def test_enum_serialization(self):
         self.assertEqual(e_new.name, e.name)
         self.assertEqual(e_new.value, e.value)
 
+    def test_tuple_serialization(self):
+        a = GoodMSONClass(a=1, b=tuple(), c=1)
+        afromdict = GoodMSONClass.from_dict(a.as_dict())
+        assert type(a.b) is tuple
```

Comment:
I think isinstance is a better approach. This applies to all instances where you use type(X) is or == something.

Reply:
isinstance checks that the instance is a tuple or a subclass of tuple. I think it is more safe to use type(a.b) is tuple as it would prevent someone (a weirdo btw ;-) to change the MSONable as_dict and/or from_dict to return a subclass of tuple when deserializing. What do you think ?

Reply:
I agree. But this is a test, not the actual code. So here, it suffices to make sure it is an instance of tuple.

Reply:
If the user serializes a tuple, he should expect to get back exactly a tuple when deserializing. Using isinstance in the test allows to change that. I could modify the MSONable from_dict to return a subclass of a tuple instead of a real tuple.